### PR TITLE
DML EP Register Identity-16

### PR DESCRIPTION
**Pull request by nums11**
This PR Registers Identity-16 to the DML EP.

ONNX Backend tests and optional type tests were skipped pending future additions.

### CODE REVIEW

File: onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc [hunk 1]
```diff
@@ -292,7 +292,21 @@
         "^test_roialign_cpu",
         "^test_dft_axis_cpu",
         "^test_dft_cpu",
-        "^test_dft_inverse_cpu"
+        "^test_dft_inverse_cpu",
```
Comment (PatriceVignola) [inline onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc:295]: So we need to add the test to both main.cc and here to disable them? Putting them only in one place doesn't work?

Comment (nums11) [reply onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc:295]: Yes

**Review by PatriceVignola — DISMISSED** [review]

**Review by PatriceVignola — APPROVED** [review]

File: onnxruntime/test/onnx/main.cc [hunk 2]
```diff
@@ -961,6 +962,21 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
     broken_tests.insert({"softmax_cross_entropy_input_shape_is_NCd1d2d3d4d5_none_no_weight_expanded", "DML does not support 5D+ tensors"});
     broken_tests.insert({"softmax_cross_entropy_input_shape_is_NCd1d2d3d4d5_none_no_weight_log_prob", "DML does not support 5D+ tensors"});
     broken_tests.insert({"softmax_cross_entropy_input_shape_is_NCd1d2d3d4d5_none_no_weight_log_prob_expanded", "DML does not support 5D+ tensors"});
+
+    // TODO: Remove identity tests when fixed #42638109
+    broken_tests.insert({"identity_cpu", "Optional type not yet supported for identity-16."});
```
Comment (fdwr) [inline onnxruntime/test/onnx/main.cc:967]: Hmm, shouldn't ORT just fall back to CPU in the case of unrecognized input/output types, just like for any other type (like bfloat16)?

Comment (fdwr) [reply onnxruntime/test/onnx/main.cc:967]: Btw @smk2007, I notice many disablements above that shouldn't be applicable anymore, like `broken_tests.insert({"unsqueeze_three_axes", "DML does not support 6D tensors"});`, as DML supports 8D. We should someday clean up the stale ones...

Comment (fdwr) [reply onnxruntime/test/onnx/main.cc:967]: Btw @smk2007, I notice many disablements above that shouldn't be applicable anymore, like `broken_tests.insert({"unsqueeze_three_axes", "DML does not support 6D tensors"});`, as DML supports 8D. We should someday clean up the stale ones...

File: onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc (hunk 1 above)
Comment (fdwr) [reply onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc:295]: @PatriceVignola : Say, if you're ever touching that area of code, I see some that can be re-enabled. e.g.

    broken_tests.insert({"unsqueeze_three_axes", "DML does not support 6D tensors"});
    broken_tests.insert({"unsqueeze_unsorted_axes", "DMLdoes not support 6D tensors"});

(we support 8D now)

File: onnxruntime/test/onnx/main.cc (hunk 2 above)
Comment (PatriceVignola) [reply onnxruntime/test/onnx/main.cc:967]: Yep, I was planning to go through them soon

**Pull request merged**